sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRoundRobinPartitioning.scala

```diff
@@ -42,21 +43,35 @@ case class GpuRoundRobinPartitioning(numPartitions: Int)
 
   def partitionInternal(batch: ColumnarBatch): (Array[Int], Array[GpuColumnVector]) = {
     val sparkTypes = GpuColumnVector.extractTypes(batch)
-    withResource(GpuColumnVector.from(batch)) { table =>
-      if (numPartitions == 1) {
+    if (numPartitions == 1) {
+      // Skip retry since partition number = 1
+      withResource(GpuColumnVector.from(batch)) { table =>
         val columns = (0 until table.getNumberOfColumns).zip(sparkTypes).map {
-          case(idx, sparkType) =>
-            GpuColumnVector.from(table.getColumn(idx).incRefCount(), sparkType)
+          case (idx, sparkType) =>
+            GpuColumnVector
+              .from(table.getColumn(idx).incRefCount(), sparkType)
         }.toArray
-        return (Array(0), columns)
+        (Array(0), columns)
       }
-      withResource(table.roundRobinPartition(numPartitions, getStartPartition)) { partedTable =>
-        val parts = partedTable.getPartitions
-        val columns = (0 until partedTable.getNumberOfColumns.toInt).zip(sparkTypes).map {
-          case(idx, sparkType) =>
-            GpuColumnVector.from(partedTable.getColumn(idx).incRefCount(), sparkType)
-        }.toArray
-        (parts, columns)
+    } else {
+      // Increase ref count since the caller will close the batch also.
+      val spillableBatch = SpillableColumnarBatch(GpuColumnVector.incRefCounts(batch),
```

Comment:
Same leak question here.

Reply:
This batch will be close in its [caller](
https://github.com/NVIDIA/spark-rapids/pull/9419/files/2dffe586c1c152c1608a5467853b0fa73176c6ba#diff-71fe5dc6abfcffa260bc813b6d1ac94645981d67e7aa763da8f8b43426312804L76).